ops/pebble.py

```diff
@@ -1054,15 +1056,20 @@ class ServiceInfo:
     def __init__(
         self,
         name: str,
-        startup: Union[ServiceStartup, str],
-        current: Union[ServiceStatus, str],
+        startup: ServiceStartup,
+        current: ServiceStatus,
     ):
         self.name = name
         self.startup = startup
         self.current = current
 
     def is_running(self) -> bool:
         """Return True if this service is running (in the active state)."""
+        if not isinstance(self.current, ServiceStatus):
+            raise ValueError(
+                'self.current must be a ServiceStatus member (e.g. ServiceStatus.ACTIVE)'
+                f', not {self.current}'
+            )
         return self.current == ServiceStatus.ACTIVE
```

Comment:
This comparison would be silently `False` if `self.current` was accidentally a `str`.

Reply:
Would a better solution be to switch to using `enum.StrEnum` instead of `enum.Enum`? This would seem to match the intent of the original code better. With `StrEnum`, `ServiceStatus.ACTIVE == 'active'`, so passing around strings instead of enum members would silently succeed instead of silently failing in the absence of a type check.

Reply:
StrEnum is a py 3.11 feature, isn't it?

We've got to support py 3.8 because Juju supports specific Ubuntu releases as bases and that's where the Python interpreter comes from.

Reply:
Good catch, @dimaqq, thanks.